pkg/client/restclient/client.go

```diff
@@ -114,6 +114,26 @@ func NewRESTClient(baseURL *url.URL, versionedAPIPath string, config ContentConf
 	}, nil
 }
 
+// WithContentConfig returns an identical RESTClient with different
+// content encoding settings.
+func (c *RESTClient) WithContentConfig(config ContentConfig) (*RESTClient, error) {
+	if config.GroupVersion == nil {
+		config.GroupVersion = &unversioned.GroupVersion{}
+	}
+	if len(config.ContentType) == 0 {
+		config.ContentType = "application/json"
+	}
+	serializers, err := createSerializers(config)
+	if err != nil {
+		return nil, err
+	}
+
+	newRC := *c
```

Comment:
This shallow copy worries me.  We're still sharing the URL pointer, the client, BackoffManager func.